### PR TITLE
Minor fix for improving support for error replies

### PR DESCRIPTION
**Pull request by rydnr**
Minor fix dealing with error replies.

Tested with AWS DocumentDB and MongoDB Atlas (TLS with no SNI).

### CODE REVIEW

File: mc/Mongo-Core.package/Mongo.class/instance/query..st [hunk 1]
```diff
@@ -5,6 +5,6 @@ query: aMongoQuery
 	aMongoQuery doBlock isNil ifTrue: [
 		aMongoQuery doBlock: [:each | results add: each]].
 	(MongoCursor root: self query: aMongoQuery ) execute.
-	(results isNotEmpty and: [ results first includesKey: '$err' ])
+	(results isNotEmpty and: [ (results first includesKey: '$err') or: [ (results first includesKey: 'errmsg') and: [ (results first includesKey: 'code') and: [ results first includesKey: 'ok' ] ] ] ])
```
Comment (tinchodias) [inline mc/Mongo-Core.package/Mongo.class/instance/query..st:8]: I'd like to know more about the new check (especially the 2 "and"s). I will write in words what I read in the code:

*Before*: We detected an error when there is a `$err` key.
*Now*: Alternatively, we will also detect an error when there are 3 keys simultaneously `errmsg`, `code` and `ok`.

Would you have some URL that documents this detection?
As a matter of fact, I don't find at this moment other info about $err than a mention [here](https://docs.mongodb.com/v4.2/reference/mongodb-wire-protocol/) in the Wire Protocol docs.

**Comment by tinchodias** [comment]
Hello @rydnr, thanks for the contribution. I added an inline review comment to learn more about the change.

Do you know what's the version of MongoDB running in AWS DocumentDB and MongoDB Atlas, where you tested it? I wonder why there is a new way to detect errors. Maybe Mongo 4.4 has news?

**Comment by rydnr** [comment]
AWS DocumentDB is not MongoDB, but supports the protocol. Or so they say.
In practice, I found some issues and ended up discarding it. But in the mean time, the errors I received from DocumentDB bypassed the existing check so I added this numb change.
It seems DocumentDB uses $errmsg instead of $err. The $code and $ok checks are there to exactly match the errors I found. For example, someone in his application uses $errmsg, as long as it doesn't use $code and $ok, this change would be harmless.

On the other hand, someone trying to use AWS DocumentDB might find this useful, if DocumentDB is suitable in his use case.

However, my motivation was two fold:
- Maybe suggest a refactor to support an extensible mechanism to detect error messages.
- Maybe include it in a set of changes to support AWS DocumentDB.
 
Since I'm not using DocumentDB, I don't need this change anymore.

**Comment by tinchodias** [comment]
> Since I'm not using DocumentDB, I don't need this change anymore.

Well, we can close the PR. It will remain as a closed PR and can be recovered in the future if needed.

> uses $errmsg, as long as it doesn't use $code and $ok

For the record, my question before was still valid anyway... I think there were missing "not"s on your code? something like: `(results first includesKey: 'code') not and: [ (results first includesKey: 'ok') not ] `.

**Pull request closed without merging**